### PR TITLE
Change `cluster.ssh` to use SkySSHRunner.

### PR DESCRIPTION
**Pull request by rohinb2**

### CODE REVIEW

**Comment by rohinb2** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/run-house/runhouse/634?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#634** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/634?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#631** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/631?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#630** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/630?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rohinb2 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: runhouse/resources/hardware/on_demand_cluster.py [hunk 1]
```diff
@@ -518,13 +519,14 @@ def ssh(self, node: str = None):
             if not sky_key.exists():
                 raise FileNotFoundError(f"Expected default sky key in path: {sky_key}")
 
+            runner = SkySSHRunner(
+                ip=node or self.address,
+                ssh_user=ssh_user,
+                port=self.ssh_port,
+                ssh_private_key=sky_key,
```
Comment (jlewitt1) [inline runhouse/resources/hardware/on_demand_cluster.py:526]: just making sure we're not causing any issues by passing in a Path object and not a string?

**Comment by rohinb2** [comment]
> 
> 
> just making sure we're not causing any issues by passing in a Path object and not a string? 

it worked so *shrug*

**Review by dongreenberg — APPROVED** [review]
LGTM!

**Comment by rohinb2** [comment]
### Merge activity

* **Mar 20, 8:34 PM EDT**: @rohinb2 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/run-house/runhouse/634).

**Pull request merged**